api/tests/unit/integrations/flagsmith/test_unit_flagsmith_client.py

```diff
@@ -0,0 +1,89 @@
+import json
+import os.path
+from unittest.mock import mock_open, patch
+
+import pytest
+
+from integrations.flagsmith.client import get_client
+
+
+@pytest.fixture(autouse=True)
+def reset_globals(mocker):
+    mocker.patch("integrations.flagsmith.client._defaults", None)
+    mocker.patch("integrations.flagsmith.client._flagsmith_client", None)
+    yield
```

Comment:
I don't _love_ this, but managing the global variables is a bit painful (but necessary I think) and I don't think it's too bad since the fixture is used by every test in this module. 

The tests in the previous iteration [here](https://github.com/Flagsmith/flagsmith/pull/2447/commits/070262b29d2de9b2b0c99a807ffa68c7e5fe8d6b#diff-1ef3c47cbe87e02aa5a53dcf72d0c31f29de4e17d9356dcbc2a982f149ca4441) were arguably cleaner, but I feel like the code itself to consume the flagsmith client was more messy. 

Part of the issue that we have is the need to be able to switch Flagsmith off entirely and rely purely on the defaults. I guess there's an argument here that we should define an option in our SDKs to run in 'offline only' mode, for which we already have an issue [here](https://github.com/Flagsmith/flagsmith/issues/2024).